File: problems/003/003.js
function solution() {
    var factor;
    var n = 600851475143;

    for (factor = 3; true; factor += 2) {
        while (n % factor === 0) {
            n /= factor;
        }

        if (factor * factor > n) {
            return n;
        } if (n === 1) {
            return factor;
        }
    }
}

if (process.argv.length === 3 && process.argv[2] === "-a") {
    console.log(solution());
    process.exit();
}

var clock = require("posix-clock");
var lazy = require("lazy");

new lazy(process.stdin).
    lines.
    forEach(function(line) {
        var iters = parseInt(line, 10);

        var start = clock.gettime(clock.MONOTONIC);
        var i;
        for (i = 0; i < iters; i++) {
            solution();
        }
        var end = clock.gettime(clock.MONOTONIC);

        var secs = end.sec - start.sec;
        var nsecs = end.nsec - start.nsec;

        console.log(secs * 1000000000 + nsecs);
    });

File: problems/003/003.lua
posix = require "posix"

local function solution()
  local factor = 1
  local n = 600851475143

  while true do
    factor = factor + 2

    while n % factor == 0 do
      n = n / factor
    end

    if factor * factor > n then
      return n
    elseif n == 1 then
      return factor
    end
  end
end

if #arg == 1 and arg[1] == "-a" then
  print(solution())
  return
end

io.stdout:setvbuf("line")
for line in io.lines() do
  local iters = tonumber(line)

  local start = { posix.clock_gettime("monotonic") }
  for i = 1,iters do
    solution()
  end
  local end_ = { posix.clock_gettime("monotonic") }

  local secs = end_[1] - start[1]
  local nsecs = end_[2] - start[2]
  print(secs * 1000000000 + nsecs)
end
